### ISSUE TITLE
Vector tiles loaded but not displayed in browser

### ISSUE THREAD

**Issue by vherbst** [urbica/martin#595]
I'm trying to load vector tiles provided by Martin using maplibre from a Postgres database. The fileserver works ok and I get the source:

```
{tilejson: "3.0.0",
tiles: ["http://localhost:3000/ln_admin_roadnetwork/{z}/{x}/{y}"],…}
bounds: [80.12214271510545, 26.366896177903985, 88.16217982661517, 30.178138014500156],
description: public.ln_admin_roadnetwork.geom",
name: "ln_admin_roadnetwork",
tilejson: "3.0.0"
tiles: ["http://localhost:3000/ln_admin_roadnetwork/{z}/{x}/{y}"]
vector_layers: [{id: "ln_admin_roadnetwork",…}]
```

I can see that tiles are loaded in the network tab from chrome. There is definitely data coming... But this layer is not displayed. Here the configuration:

```
  map.addLayer({
          id: 'ln_admin_roadnetwork',
          type: 'line',
          source: {
            type: 'vector',
            url: 'http://localhost:3000/ln_admin_roadnetwork'
          },
          'source-layer': 'ln_admin_roadnetwork',
          paint: {
            'line-color': '#fd0000',
            'line-width': 2
          },
        });
```

nothing complicated. Another layer from an example works fine:

```
map.addSource('contours', {
          type: 'vector',
          url: 'https://api.maptiler.com/tiles/contours/tiles.json?key=get_your_own_OpIi9ZULNHzrESv6T2vL'
        });
 map.addLayer({
          'id': 'terrain-data',
          'type': 'line',
          'source': 'contours',
          'source-layer': 'contour',
          'layout': {
            'line-join': 'round',
            'line-cap': 'round'
          },
          'paint': {
            'line-color': '#ff69b4',
            'line-width': 1
          }
        });
```

This one works and I can see data. Why is the first one not appearing on the map even so it is loaded in the browser?
Thank you very much...

**Comment by nyurik** [comment]
What type of geometries do you have in your table?  Point geometries might not work with `'line-*'` params

**Comment by vherbst** [comment]
It's MultiLine and shows up in Qgis all right. Now I tried a new point layer (a view) with srid 4326 type point. 

```
geometry_colums: 
dssgis | public | irrigation_test | geom | 2 | 4326 | POINT
```

here the whole src for loading:

```
import maplibre from 'maplibre-gl';
import { shallowRef, onMounted, onUnmounted, markRaw } from 'vue';
import * as os from "os";
export default {
  name: "Map",
  setup: function () {

    onMounted(() => {
      const map = markRaw(new maplibre.Map({
        container: 'map',
        style: osstyle, // 'https://demotiles.maplibre.org/style.json', // stylesheet location
        center: [84, 28], // starting position [lng, lat]
        zoom: 6 // starting zoom
      }));
      map.addControl(new maplibre.NavigationControl());
      new maplibre.Marker({color: "#FF0000"})
          .setLngLat([139.7525, 35.6841])
          .addTo(map);
      const markerPopup = new maplibre.Popup().setText('Some marker');
      new maplibre.Marker().setLngLat([84, 28]).setPopup(markerPopup).addTo(map);
      map.on('load', function() {
        map.addLayer({
          id: 'terrain-data',
          type: 'line',
          source: {
            type: 'vector',
            url: 'https://api.maptiler.com/tiles/contours/tiles.json?key=get_your_own_OpIi9ZULNHzrESv6T2vL'
          },
          'source-layer': 'contour',
          layout: {
            'line-join': 'round',
            'line-cap': 'round'
          },
          paint: {
            'line-color': '#ff69b4',
            'line-width': 1
          }
        })
        map.addLayer({
          id: 'irrigation_test',
          type: 'circle',
          visibility: 'visible',
          source: {
            type: 'vector',
            url: 'http://localhost:3000/irrigation_test'
          },
          'source-layer': 'irrigation_test',
          paint: {
            'circle-color': '#ff69b4',
            'circle-radius' : 20
          },
        });

      });
    
    });
    
  }}

const osstyle = {
  "version": 8,
  "sources": {
    "osm": {
      "type": "raster",
      "tiles": ["https://a.tile.openstreetmap.org/{z}/{x}/{y}.png"],
      "tileSize": 256,
      "attribution": "&copy; OpenStreetMap Contributors",
      "maxzoom": 19
    }
  },
  "layers": [
    {
      "id": "osm",
      "type": "raster",
      "source": "osm" // This must match the source key above
    }
  ]
};
```

Here a view of the browser:

Anfrage-URL: http://localhost:3000/irrigation_test/8/187/106
Anfragemethode: GET
Statuscode: 200 OK
Remote-Adresse: 127.0.0.1:3000
Richtlinien für Verweis-URL: strict-origin-when-cross-origin

and the data

```
�
public.irrigation_test.geom��"	�+�A��"	�>�@��"	��@��"	��@��"	�-�@��"	�>�@��"	��@��"	
Karnali"(�(� x ......
```

So why is nothing displayed on the map? I just don't know where to look now...

**Comment by vherbst** [comment]
One more thing I noticed: I could download the vector tiles from the browser and display them in Qgis. The only difference from tiles which display is that the name is without the extension .pbf. So all vector tiles from other sources have the ending pbf and from Martin have no pbf ending.

**Comment by nyurik** [comment]
One thing i noticed is that your tile data has `public.irrigation_test.geom` -- this is suspicious, because I would expect this to be the layer ID, not the description. What configuration did you use with Martin? Is it auto-generated one, or manual config?  I wonder if you need to change your javascript code to use that string as an ID.

P.S. Please use 3 tick symbols when creating block of code, not 2 -- "```"

**Comment by vherbst** [comment]
ok the only way I got it to work without errors in the browser was to specify the id including the geom column:
```
  tables:
    public.ln_admin_boundary.geom:
      schema: public
      table: ln_admin_boundary
      srid: 4326
      geometry_column: geom
      id_column: gid
      minzoom: 5
      maxzoom: 30
      bounds: [80.058, 26.34, 88.84, 30.59]
      extent: 4096
      buffer: 64
      clip_geom: true
      geometry_type: MULTILINESTRING
      properties:
        bndid: int4
        bndtype: varchar
        gid: int4

```

und in javascript code:

```
 map.addLayer({
          id: 'public.ln_admin_boundary.geom',
          type: 'line',
          source: {
            type: 'vector',
            url: 'http://localhost:3000/public.ln_admin_boundary.geom'
          },
          'source-layer': 'public.ln_admin_boundary.geom',
          paint: {
            'line-color': '#ff69b4',
            'line-width': 2
          }
        });

```

all other combinations with names resulted in not displaying or raising errors in the browser. I think this naming was not intended?

**Comment by nyurik** [comment]
@vherbst of course -- the layer ID will match the ID of the source itself:

```
  tables:
    public.ln_admin_boundary.geom:     <--------- this source ID will be the layer ID
      schema: public
      table: ln_admin_boundary
```

I do think we should make this configurable though.  I will create a new issue for that.

P.S. note that the source ID could be anything else according to the source ID rules from the readme

**Comment by vherbst** [comment]
But in the documentation it is not stated that geometry column name (geom) has to be added to the layer source name AND url. There was no way to make it work without geom at the end (I played around with the config quite a while) , hence the generated config did not work. Don't know if it is a problem only on my side or a bug in maplibre.

**Comment by nyurik** [comment]
@vherbst I am not sure i understand.  You should be able to use this config:

```
  tables:
    foobar:   // <<<<------  custom source ID
      schema: public
      table: ln_admin_boundary
      srid: 4326
      geometry_column: geom
      id_column: gid
     ...
```

and then in your javascript use the `foobar`, e.g.

```js
 map.addLayer({
          id: 'foobar',
          type: 'line',
          source: {
            type: 'vector',
            url: 'http://localhost:3000/foobar'
          },
          'source-layer': 'foobar',
          paint: {
            'line-color': '#ff69b4',
            'line-width': 2
          }
        });
```

**Comment by gtitov** [comment]
I face the same issue with **martin v0.7.2** trying to get layers in **maplibre-gl@2.4.0** and **mapbox-gl-js v2.9.1**.

Rolling back to **martin v0.5.0** solved the issue

**Comment by nyurik** [comment]
@gtitov please provide more details - Martin 0.7 has made significant changes and improvements to the API, paths, and config structure - you might need to update your config and maplibre code.

**Comment by gtitov** [comment]
@nyurik sure

Public ip will work for some time for you to check if it helps

I run martin with a connection string like [this](https://github.com/maplibre/martin#usage) `martin postgresql://postgres@localhost/db`

Just like @vherbst said the map fetches tiles as I check it in Network tab but don't render. I wonder if there is a problem with `source-layer` in `map.addLayer` because if I use `description` rather than `vector_layers:[{id}]` or `name` from a tilejson a map renders the layer but throughs an error in a browser console.

Here is the code for **martin v0.5.0** _(no error, renders layer)_

```js
map.addSource("cities5", {
    "type": "vector",
    "url": "http://213.219.215.248:3005/public.ne_10m_populated_places.json"
})

map.addLayer({
    "id": "cities5-layer",
    "type": "circle",
    "source": "cities5",
    "source-layer": "public.ne_10m_populated_places",
    "paint": {
        "circle-stroke-width": 1,
        "circle-stroke-color": "#ffffff",
        "circle-color": "#1a9641",
        "circle-opacity": 0.8,
        "circle-radius": 8
    }
})
```

the the code for **martin v0.7.2** using `id` from `vector_layers` the tilejson _(no error, no layer)_

```js
map.addSource("cities7", {
    "type": "vector",
    "url": "http://213.219.215.248:3007/ne_10m_populated_places"
})

map.addLayer({
    "id": "cities7-layer",
    "type": "circle",
    "source": "cities7",
    "source-layer": "ne_10m_populated_places",
    "paint": {
        "circle-stroke-width": 1,
        "circle-stroke-color": "#ffffff",
        "circle-color": "#ff0000",
        "circle-opacity": 0.8,
        "circle-radius": 4
    }
})
```

the the code for **martin v0.7.2** using `description` from the tilejson _(throughs error, renders layer)_

```js
map.addSource("cities7", {
    "type": "vector",
    "url": "http://213.219.215.248:3007/ne_10m_populated_places"
})

map.addLayer({
    "id": "cities7-layer",
    "type": "circle",
    "source": "cities7",
    "source-layer": "ne_10m_populated_places",
    "paint": {
        "circle-stroke-width": 1,
        "circle-stroke-color": "#ffffff",
        "circle-color": "#0000ff",
        "circle-opacity": 0.8,
        "circle-radius": 2
    }
})
```

```
Error: Source layer "public.ne_10m_populated_places.geom" does not exist on source "cities7" as specified by style layer "cities7-layer-2"
```

### Reproducible example

```html
<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">

    <script src='https://unpkg.com/maplibre-gl@2.4.0/dist/maplibre-gl.js'></script>
    <link href='https://unpkg.com/maplibre-gl@2.4.0/dist/maplibre-gl.css' rel='stylesheet' />

    <title>Document</title>
</head>

<body>
    <div id="map" style="position: absolute; top: 0; bottom: 0; width: 100%; z-index: -1"></div>
    <script>
        document.addEventListener("DOMContentLoaded", function () {
            var map = new maplibregl.Map({
                container: "map",
                style: {
                    version: 8,
                    sources: {},
                    layers: []
                },
                center: [37, 55],
                zoom: 3
            })

            map.on("load", function () {
                map.addSource("cities5", {
                    "type": "vector",
                    "url": "http://213.219.215.248:3005/public.ne_10m_populated_places.json"
                })

                map.addLayer({
                    "id": "cities5-layer",
                    "type": "circle",
                    "source": "cities5",
                    "source-layer": "public.ne_10m_populated_places",
                    "paint": {
                        "circle-stroke-width": 1,
                        "circle-stroke-color": "#ffffff",
                        "circle-color": "#1a9641",
                        "circle-opacity": 0.8,
                        "circle-radius": 8
                    }
                })

                map.addSource("cities7", {
                    "type": "vector",
                    "url": "http://213.219.215.248:3007/ne_10m_populated_places"
                })

                map.addLayer({
                    "id": "cities7-layer",
                    "type": "circle",
                    "source": "cities7",
                    "source-layer": "ne_10m_populated_places",
                    "paint": {
                        "circle-stroke-width": 1,
                        "circle-stroke-color": "#ffffff",
                        "circle-color": "#ff0000",
                        "circle-opacity": 0.8,
                        "circle-radius": 4
                    }
                })

                map.addLayer({
                    "id": "cities7-layer-2",
                    "type": "circle",
                    "source": "cities7",
                    "source-layer": "public.ne_10m_populated_places.geom",
                    "paint": {
                        "circle-stroke-width": 1,
                        "circle-stroke-color": "#ffffff",
                        "circle-color": "#0000ff",
                        "circle-opacity": 0.8,
                        "circle-radius": 2
                    }
                })
            })
        })
    </script>
</body>

</html>
```

**Comment by nyurik** [comment]
Thanks @vherbst and @gtitov, you are right - this was an accidental change in functionality.  I fixed it in #598, will release soon as 0.8 version.

**Comment by nyurik** [comment]
this feature has been released in 0.8.0
